### PR TITLE
IGNITE-23107 Add REST API for migrating running nodes missed during CMG repair

### PR DESCRIPTION
**Pull request by rpuch**
https://issues.apache.org/jira/browse/IGNITE-23107

### CODE REVIEW

File: modules/rest-api/src/main/java/org/apache/ignite/internal/rest/api/recovery/system/MigrateRequest.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    @Schema(description = "Ignite version.")
+    private final String version;
+
+    @Schema(description = "ID of the cluster.")
+    private final UUID clusterId;
+
+    @Schema(description = "Name of the cluster.")
+    private final String clusterName;
+
+    @Schema(description = "IDs the cluster had before.")
```
Comment (tkalkirill) [inline modules/rest-api/src/main/java/org/apache/ignite/internal/rest/api/recovery/system/MigrateRequest.java:51]: Indicate what `null` means.

File: modules/rest-api/src/main/java/org/apache/ignite/internal/rest/api/recovery/system/MigrateRequest.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        return clusterId;
+    }
+
+    /** Returns name of the cluster. */
+    @JsonGetter("clusterName")
+    public String clusterName() {
+        return clusterName;
+    }
+
+    /** Returns IDs the cluster had before. */
```
Comment (tkalkirill) [inline modules/rest-api/src/main/java/org/apache/ignite/internal/rest/api/recovery/system/MigrateRequest.java:108]: Indicate what `null` means.

File: modules/rest-api/src/main/java/org/apache/ignite/internal/rest/api/recovery/system/SystemDisasterRecoveryApi.java [hunk 3]
```diff
@@ -51,6 +51,20 @@ public interface SystemDisasterRecoveryApi {
...
+    @Produces({MediaType.APPLICATION_JSON, MediaType.PROBLEM_JSON})
     CompletableFuture<Void> reset(@Body ResetClusterRequest command);
+
+    @Post("migrate")
+    @Operation(
+            operationId = "migrate",
+            description = "Migrates nodes from old cluster to new (repaired) cluster."
+    )
+    @ApiResponse(responseCode = "200", description = "Migration initiated.")
+    @ApiResponse(responseCode = "500", description = "Internal error.",
```
Comment (tkalkirill) [inline modules/rest-api/src/main/java/org/apache/ignite/internal/rest/api/recovery/system/SystemDisasterRecoveryApi.java:63]: There are quite a lot of `description = "Internal error."` in the project, maybe we can turn it into a constant with a separate ticket, what do you think?

File: modules/rest-api/src/main/java/org/apache/ignite/internal/rest/api/recovery/system/SystemDisasterRecoveryApi.java [hunk 4]
```diff
@@ -51,6 +51,20 @@ public interface SystemDisasterRecoveryApi {
...
+            content = @Content(mediaType = MediaType.PROBLEM_JSON, schema = @Schema(implementation = Problem.class)))
+    @ApiResponse(responseCode = "400", description = "Bad request.",
```
Comment (tkalkirill) [inline modules/rest-api/src/main/java/org/apache/ignite/internal/rest/api/recovery/system/SystemDisasterRecoveryApi.java:65]: Same about constant

File: modules/rest-api/src/main/java/org/apache/ignite/internal/rest/exception/handler/MigrateExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import io.micronaut.http.server.exceptions.ExceptionHandler;
+import jakarta.inject.Singleton;
+import org.apache.ignite.internal.disaster.system.exception.MigrateException;
+import org.apache.ignite.internal.rest.api.Problem;
+import org.apache.ignite.internal.rest.constants.HttpCode;
+import org.apache.ignite.internal.rest.problem.HttpProblemResponse;
+
+/**
+ * Handles {@link MigrateException} and represents it as a rest response.
+ */
```
Comment (tkalkirill) [inline modules/rest-api/src/main/java/org/apache/ignite/internal/rest/exception/handler/MigrateExceptionHandler.java:32]:
```suggestion
/** Handles {@link MigrateException} and represents it as a rest response. */
```

File: modules/rest-api/src/main/java/org/apache/ignite/internal/rest/api/recovery/system/SystemDisasterRecoveryApi.java (hunk 3 above)
Comment (rpuch) [reply modules/rest-api/src/main/java/org/apache/ignite/internal/rest/api/recovery/system/SystemDisasterRecoveryApi.java:65]: Added a ticket https://issues.apache.org/jira/browse/IGNITE-23274 and TODOs

**Review by tkalkirill — APPROVED** [review]

**Pull request merged**